homeassistant/components/nightscout/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""The Nightscout integration."""
+import asyncio
+from asyncio import TimeoutError as AsyncIOTimeoutError
+import logging
+
+from aiohttp import ClientError
+from py_nightscout import Api as NightscoutAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import SLOW_UPDATE_WARNING
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["sensor"]
+_API_TIMEOUT = SLOW_UPDATE_WARNING - 1
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Nightscout component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Nightscout from a config entry."""
+    server_url = entry.data[CONF_URL]
+
+    api = NightscoutAPI(server_url)
```

Comment:
We want the libraries that use aiohttp to use the home assistant aiohttp client session. Please get the home assistant session and pass it to the client api when instantiating that.

https://github.com/home-assistant/core/blob/4c8c8a8c999afc1048349d2627c4aef85b812b50/homeassistant/helpers/aiohttp_client.py#L32-L37